lib/autoupdate/rollout/strategy.go

```diff
@@ -94,3 +94,35 @@ func setGroupState(group *autoupdate.AutoUpdateAgentRolloutStatusGroup, newState
 		group.LastUpdateTime = timestamppb.New(now)
 	}
 }
+
+func computeRolloutState(groups []*autoupdate.AutoUpdateAgentRolloutStatusGroup) autoupdate.AutoUpdateAgentRolloutState {
+	groupCount := len(groups)
+
+	if groupCount == 0 {
+		return autoupdate.AutoUpdateAgentRolloutState_AUTO_UPDATE_AGENT_ROLLOUT_STATE_UNSPECIFIED
+	}
+
```

Comment:
Please do it something like

```suggestion
var done, unstarted int
for _, group := range groups {
  switch group.State {
  case ROLLEDBACK: return ROLLEDBACK
  case DONE: done++
  case UNSTARTED: unstarted++
  }
}

if done == groupCount { return DONE }
if unstarted == groupCount { return UNSTARTED }
return ACTIVE
}

```